src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsUrlTest.java

```diff
@@ -0,0 +1,197 @@
+                checksNamesMap.put(dirName, fileNames);
+            }
+            final String[] miscNames = checks.list(filter);
+            checksNamesMap.put(MISC, miscNames);
+        }
+        return checksNamesMap;
+    }
+
+    @Test
+    public void testXdocsUrl() throws Exception {
+        final SAXParserFactory parserFactory = SAXParserFactory.newInstance();
+        final SAXParser parser = parserFactory.newSAXParser();
+        final CheckTest checkHandler = new CheckTest(NODE_NAME);
+        final List<List<String>> list;
+        try (InputStream input = Files.newInputStream(AVAILABLE_CHECKS_PATH)) {
+            parser.parse(input, checkHandler);
+            list = checkHandler.list;
+        }
+        final Map<String, String[]> checksNamesMap = getXdocsMap();
+        for (List<String> sub : list) {
+            final String moduleName = sub.get(0);
+            final String checkName1 = sub.get(1);
+            final String checkName2 = sub.get(2);
+            final String errorMessage =
+                    "URL in" + AVAILABLE_CHECKS_PATH + " is inconsistent: "
+                            + moduleName + " " + checkName1 + " " + checkName2;
+            assertEquals(checkName1, checkName2, errorMessage);
+            if (checkName1.equals(COMMENTS_INDENTATION) || checkName1.equals(INDENTATION)) {
+                assertEquals(MISC, moduleName, errorMessage);
+            }
+            else if (checkName1.equals(SUPPRESS_WARNINGS_HOLDER)) {
+                assertEquals(ANNOTATION, moduleName, errorMessage);
+            }
+            else {
+                final String[] moduleFileNames = checksNamesMap.get(moduleName);
+                if (moduleFileNames != null) {
+                    boolean match = false;
+                    for (String filename : moduleFileNames) {
+                        if ((checkName1 + SUFFIX_CHECK_JAVA).equals(filename)) {
+                            match = true;
+                            break;
+                        }
+                    }
+                    assertTrue(match, errorMessage);
+                }
+            }
+        }
+    }
+
+    public static final class CheckTest extends DefaultHandler {
+
+        public static final String SPLIT_ONE = "#";
+
+        public static final String SPLIT_TWO = "\\.";
+
+        public static final String PREFIX_CONFIG = "config_";
+
+        private final String nodeName;
+
+        private List<List<String>> list;
+
+        private List<String> stringList;
+
+        private String currentTag;
+
+        public CheckTest(String nodeName) {
+            this.nodeName = nodeName;
+        }
+
+        public List<List<String>> getList() {
+            final List<List<String>> copyList = new ArrayList<>();
+            for (List<String> elemList: list) {
+                final List<String> tempList = new ArrayList<>(elemList);
+                copyList.add(tempList);
+            }
+            return copyList;
```

Comment:
why you need to create copy of list?
I see no usages of this getList() method in code

Reply:
done, remove it, it is redundant